src/daemon/daemon.cpp

```diff
@@ -801,6 +804,70 @@ std::string generate_unused_mac_address(std::unordered_set<std::string>& s)
                     max_tries, s.size())};
 }
 
+std::string run_in_vm(mp::SSHSession& session, const std::string& cmd)
+{
+    auto proc = session.exec(cmd);
+    if (proc.exit_code() != 0)
+    {
+        auto error_msg = proc.read_std_error();
+        mpl::log(mpl::Level::warning, category,
+                 fmt::format("failed to run '{}', error message: '{}'", cmd, mp::utils::trim_end(error_msg)));
+        return std::string{};
+    }
+
+    auto output = proc.read_std_output();
+    if (output.empty())
+    {
+        mpl::log(mpl::Level::warning, category, fmt::format("no output after running '{}'", cmd));
+        return std::string{};
+    }
+
+    return mp::utils::trim_end(output);
+}
+
+bool is_ipv4_valid(const std::string& ipv4)
+{
+    try
+    {
+        (mp::IPAddress(ipv4));
+    }
+    catch (std::invalid_argument&)
+    {
+        return false;
+    }
+
+    return true;
+}
+
+std::vector<std::string> get_all_ipv4(mp::SSHSession& session)
+{
+    std::vector<std::string> all_ipv4;
+
+    auto ip_a_output = QString::fromStdString(run_in_vm(session, "ip a"));
+
+    QRegularExpression ipv4_re{QStringLiteral("^\\s+inet ([\\d\\.]+)/\\d+ .*scope ([a-z]+)[ \\n]"),
+                               QRegularExpression::MultilineOption};
+
+    QRegularExpressionMatchIterator ip_it = ipv4_re.globalMatch(ip_a_output);
+
+    while (ip_it.hasNext())
+    {
+        auto ip_match = ip_it.next();
+        auto scope = ip_match.captured(2);
+        auto ip = ip_match.captured(1).toStdString();
+
+        if (scope == "global" || scope == "link")
+        {
+            mpl::log(mpl::Level::debug, category, fmt::format("adding ip {} with scope {}", ip, scope));
```

Comment:
Good point. I removed them, because in fact these logs are leftovers from the coding stage, when I was seeing if the IP's were correctly added or not to the list.